packages/preset-arbitrary/src/split-arbitrary.ts

```diff
@@ -0,0 +1,37 @@
+import type { Extractor } from '@unocss/core'
+import { defaultSplitRE, isValidSelector } from '@unocss/core'
+
+export const quotedArbitraryValuesRE = /(?:[\w&:[\]-]|\[\S+=\S+\])+\[\\?['"]?\S+?['"]\]\]?[\w:-]*/g
+export const arbitraryPropertyRE = /\[(\\\W|[\w-])+:[^\s:]*?("\S+?"|'\S+?'|`\S+?`|[^\s:]+?)[^\s:]*?\)?\]/g
+const arbitraryPropertyCandidateRE = /^\[(\\\W|[\w-])+:['"]?\S+?['"]?\]$/
+
+export function splitCode(code: string) {
+  const result = new Set<string>()
+
+  for (const match of code.matchAll(arbitraryPropertyRE)) {
+    if (!code[match.index! - 1]?.match(/^[\s'"`]/))
+      continue
+
+    result.add(match[0])
+  }
+
+  for (const match of code.matchAll(quotedArbitraryValuesRE))
+    result.add(match[0])
+
+  code
+    .split(defaultSplitRE)
+    .forEach((match) => {
+      if (isValidSelector(match) && !arbitraryPropertyCandidateRE.test(match))
+        result.add(match)
+    })
```

Comment:
This probably will not work as it was working previously in `split.ts`. The purpose of these lines is to reduce some false positives captured by `defaultSplitRE`. We remove those false positives on line 12.

If I remember correctly, this is capturing false positives like that:

```
hover="[&>span]:text-white"
```

Which is excluded from the tests currently. 

`split.ts` will capture `[&>span]:text-white` and presetAttributify will capture `hover="[&>span]:text-white"` and unocss will generate unused `[&>span]:text-white`.

If this is a thing, we can't merge this PR without resolving this regression.

Reply:
Maybe we should provide `code` variable as a MagicString instance and remove the matched utilities from the code. That means this extractor should run before the default splitter (split.ts).

Reply:
Another solution is would be removing the default splitter from the extractors list since this extractor already does its job. This is not an ideal solution and we don't have access to uno config on here as of now.